File: Code/Chapter 8/C#/EditData/Form1.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using CreateDataSet;
#endregion

namespace EditData
{
   partial class Form1 : Form
   {
      private CustProd myDataSet;

      public Form1()
      {
         InitializeComponent();
      }

      private void btnCreate_Click(object sender, EventArgs e)
      {
         myDataSet = new CustProd();
         dgCustomers.DataSource = myDataSet.Customers;
         dgCustomerProducts.DataSource = myDataSet.CustomerProducts;
         dgProducts.DataSource = myDataSet.Products;
      }

      private void btnLoad_Click(object sender, EventArgs e)
      {
         OpenFileDialog dlg = new OpenFileDialog();
         dlg.Filter = "Xml Files (*.xml)|*.xml"; ;
         dlg.ShowDialog();
         if (dlg.FileName.Length > 0)
            LoadXML(dlg.FileName);
      }

      private void btnSave_Click(object sender, EventArgs e)
      {
         SaveFileDialog dlg = new SaveFileDialog();
         dlg.Filter = "Xml Files (*.xml)|*.xml"; ;
         dlg.ShowDialog();
         if (dlg.FileName.Length > 0)
            myDataSet.WriteXml(dlg.FileName);
      }

      private void LoadXML(string fileName)
      {
         myDataSet = new CustProd();
         myDataSet.ReadXml(fileName);
         dgCustomers.DataSource = myDataSet.Customers;
         dgCustomerProducts.DataSource = myDataSet.CustomerProducts;
         dgProducts.DataSource = myDataSet.Products;
      }

      private void btnClose_Click(object sender, EventArgs e)
      {
         this.Close();
      }
   }
}

File: Code/Chapter 8/VB.NET/EditData/Form1.vb
Imports CreateDataSet

Public Class Form1

    Private myDataSet As CustProd

    Private Sub btnLoad_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnLoad.Click
        Dim dlg As OpenFileDialog = New OpenFileDialog()
        dlg.Filter = "Xml Files (*.xml)|*.xml"
        dlg.ShowDialog()
        If dlg.FileName.Length > 0 Then
            LoadXML(dlg.FileName)
        End If
    End Sub
    Private Sub btnCreate_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCreate.Click
        myDataSet = New CustProd()
        dgCustomers.DataSource = myDataSet.Customers
        dgCustomerProducts.DataSource = myDataSet.CustomerProducts
        dgProducts.DataSource = myDataSet.Products
    End Sub
    Private Sub btnSave_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSave.Click
        Dim dlg As SaveFileDialog = New SaveFileDialog()
        dlg.Filter = "Xml Files (*.xml)|*.xml"
        dlg.ShowDialog()
        If dlg.FileName.Length > 0 Then
            myDataSet.WriteXml(dlg.FileName)
        End If
    End Sub
    Private Sub btnClose_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnClose.Click
        Me.Close()
    End Sub

    Private Sub LoadXML(ByVal fileName As String)
        myDataSet = New CustProd()
        myDataSet.ReadXml(fileName)
        dgCustomers.DataSource = myDataSet.Customers
        dgCustomerProducts.DataSource = myDataSet.CustomerProducts
        dgProducts.DataSource = myDataSet.Products
    End Sub
End Class
